server/backend/db_backend.go

```diff
@@ -238,6 +247,15 @@ func (self *MongoBackend) importTx(ctx context.Context, tx *types.Transaction, b
 		return fmt.Errorf("failed to upsert tx: %v", err)
 	}
 
+	err = self.insertTransactionsByAddress(ctx, transaction.From, transaction.TxHash, transaction.CreatedAt)
+	if err != nil {
+		return err
+	}
+	err = self.insertTransactionsByAddress(ctx, transaction.To, transaction.TxHash, transaction.CreatedAt)
```

Comment:
What if `to == from`? What if `to == 0x0`?

Reply:
to==from - only one transaction would be referred (we're using this collection only for extracting IDs nothing else) - there is upsert and uniq index on tx_hash + address

to==0x0 - fixed

Reply:
I know it will just overwrite and not produce a duplicate, but why make the extra call at all, when we could just do a simple check for `to == from` and skip when it's unnecessary?

Reply:
makes sense